azurerm/resource_arm_metric_alertrule.go

```diff
@@ -276,7 +276,7 @@ func resourceArmMetricAlertRuleRead(d *schema.ResourceData, meta interface{}) er
 		d.Set("webhook_action", webhook_actions)
 	}
 
-	flattenAndSetTags(d, resp.Tags)
+	flattenAndSetTags(d, resp.Tags, "$type")
```

Comment:
this function's already doing too much as it is - can we filter out the tags to remove before passing this in? e.g.

```
tagsToOutput := make([]string, 0)
for k, v := range resp.Tags {
  if strings.EqualFold(k, "$type") {
   continue
  }
  tagsToOutput[k] = v
}
flattenAndSetTags(d, tagsToOutput, "$type")
```

Reply:
The reason that I refactored the function is to avoid future copy & paste for other resources:
1. The effort to add filtering logic is pretty small(few lines of code).
2. It does not break the KISS principle but emphasize the DRY principle, it's very common that having a customized filtering before doing any real work.